src/Init/NotationExtra.lean

```diff
@@ -170,16 +170,23 @@ See [Theorem Proving in Lean 4][tpil4] for more information.
 -/
 syntax (name := calcTactic) "calc" calcSteps : tactic
 
+/-- Denotes a term that was omitted from the delaborator output due to `pp.deepTerms false`. -/
+syntax "⋯" : term
+
 @[app_unexpander Unit.unit] def unexpandUnit : Lean.PrettyPrinter.Unexpander
   | `($(_)) => `(())
 
 @[app_unexpander List.nil] def unexpandListNil : Lean.PrettyPrinter.Unexpander
   | `($(_)) => `([])
 
 @[app_unexpander List.cons] def unexpandListCons : Lean.PrettyPrinter.Unexpander
-  | `($(_) $x [])      => `([$x])
-  | `($(_) $x [$xs,*]) => `([$x, $xs,*])
-  | _                  => throw ()
+  | `($(_) $x $tail) =>
+    match tail with
+    | `([])      => `([$x])
+    | `([$xs,*]) => `([$x, $xs,*])
+    | `(⋯)       => `([$x, $tail]) -- Unexpands to `[x, y, z, ⋯]` for `⋯ : List α`
+    | _          => throw ()
+  | _ => throw ()
```

Comment:
For long lists, I guess this unexpander has been yielding quadratic complexity. Maybe eventually we should have a delaborator instead, so it can allocate the array all at once. That could also implement Joachim's suggestion to show the first and last elements of a long list. That's not for this PR though.